x-pack/plugins/ml/public/alerting/anomaly_detection_alerts_table/alert_actions.tsx

```diff
@@ -0,0 +1,210 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  EuiButtonIcon,
+  EuiContextMenuItem,
+  EuiContextMenuPanel,
+  EuiPopover,
+  EuiToolTip,
+} from '@elastic/eui';
+
+import React, { useCallback, useMemo, useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import { CaseAttachmentsWithoutOwner } from '@kbn/cases-plugin/public';
+import { AttachmentType } from '@kbn/cases-plugin/common';
+import { EcsSecurityExtension as Ecs } from '@kbn/securitysolution-ecs';
+import {
+  ALERT_RULE_NAME,
+  ALERT_RULE_UUID,
+  ALERT_STATUS,
+  ALERT_STATUS_ACTIVE,
+  ALERT_UUID,
+} from '@kbn/rule-data-utils';
+import { useBulkUntrackAlerts } from '@kbn/triggers-actions-ui-plugin/public';
+import { type Alert } from '@kbn/triggers-actions-ui-plugin/public/types';
+import { PLUGIN_ID } from '../../../common/constants/app';
+import { useMlKibana } from '../../application/contexts/kibana';
+
+export interface AlertActionsProps {
+  alert: Alert;
+  ecsData: Ecs;
+  id?: string;
+  refresh: () => void;
+  setFlyoutAlert: React.Dispatch<React.SetStateAction<any | undefined>>;
+}
+
+export function AlertActions({
+  alert,
+  ecsData,
+  id: pageId,
+  refresh,
+  setFlyoutAlert,
+}: AlertActionsProps) {
+  const {
+    cases,
+    http: {
+      basePath: { prepend },
+    },
+  } = useMlKibana().services;
+  const casesPrivileges = cases?.helpers.canUseCases();
+
+  const { mutateAsync: untrackAlerts } = useBulkUntrackAlerts();
+
+  const [isPopoverOpen, setIsPopoverOpen] = useState<boolean>(false);
+
+  const ruleId = alert[ALERT_RULE_UUID]?.[0] ?? null;
+  const alertId = alert[ALERT_UUID]?.[0] ?? '';
+
+  const linkToRule = ruleId
+    ? prepend(`/app/management/insightsAndAlerting/triggersActions/rule/${ruleId}`)
+    : null;
+
+  const caseAttachments: CaseAttachmentsWithoutOwner = useMemo(() => {
+    return ecsData?._id
+      ? [
+          {
+            alertId: alertId ?? '',
+            index: ecsData?._index ?? '',
+            type: AttachmentType.alert,
+            rule: {
+              id: ruleId,
+              name: alert[ALERT_RULE_NAME]![0],
+            },
+            owner: PLUGIN_ID,
+          },
+        ]
+      : [];
+  }, [alert, alertId, ecsData?._id, ecsData?._index, ruleId]);
+
+  const isActiveAlert = useMemo(() => alert[ALERT_STATUS]![0] === ALERT_STATUS_ACTIVE, [alert]);
+
+  const onSuccess = useCallback(() => {
+    refresh();
+  }, [refresh]);
+
+  const createCaseFlyout = cases!.hooks.useCasesAddToNewCaseFlyout({ onSuccess });
+  const selectCaseModal = cases!.hooks.useCasesAddToExistingCaseModal({ onSuccess });
+
+  const closeActionsPopover = () => {
+    setIsPopoverOpen(false);
+  };
+
+  const toggleActionsPopover = () => {
+    setIsPopoverOpen(!isPopoverOpen);
+  };
+
+  const handleAddToNewCaseClick = () => {
```

Comment:
Is there something wrong with the permissions check here? I am able to see connectors, for example, when adding the swim lane to a new case.

<img width="668" alt="image" src="https://github.com/elastic/kibana/assets/7405507/0aa23144-fb2d-462f-bddc-1529af2afdaf">

Reply:
@XavierM do you have an idea what might be an issue here?

Reply:
@cnasikas should be the right person to answer here

Reply:
I think this is happening because the `triggers_actions_ui` plugin wraps the alert table with the `CasesContext` and [sets the permissions](https://github.com/elastic/kibana/blob/5230db48651e5dc5566d480c762c701e2a1fb99a/x-pack/plugins/triggers_actions_ui/public/application/sections/alerts_table/alerts_table_state.tsx#L279) without an owner (empty array). Because there is no owner all permissions are set to `false`. The inner  `CasesContext` cancels the outer `CasesContext` set by the ML plugin. if [in](https://github.com/elastic/kibana/blob/5230db48651e5dc5566d480c762c701e2a1fb99a/x-pack/plugins/triggers_actions_ui/public/application/sections/alerts_table/alerts_table_state.tsx#L279) you pass `undefined` the permissions will return the authorized one for all available owners. @XavierM This is a bug in our side. Ideally, we should decouple the `CasesContext` from the alerts table and let the consumers wrap the table with the `CasesContext`. Is this feasible? Wdyt?

Reply:
I think that will be extremely annoying for the user, let m eplay with it to see if I understood everything here

Reply:
@cnasikas Watching this conversation, cause we are currently trying to integrate the alerts table as an embeddable and I faced some issues when interacting with the Actions (3 dots) and adding to cases. Ideally we want to disable the actions. @XavierM Is this possible? Don't want to abuse this thread, I will separately ping you guys.

<img width="1155" alt="Screenshot 2023-11-09 at 00 07 37" src="https://github.com/elastic/kibana/assets/2852703/d1989730-3699-4680-a3ff-18f5f98f3900">

Reply:
@XavierM I think both approaches have pros and cons. Let's talk about it offline.